picard/util/pipe.py

```diff
@@ -190,7 +194,7 @@ def __generate_filenames(self, app_name: str, app_version: str) -> List[str]:
         for directory in self.PIPE_DIRS:
             if directory:
                 _pipe_names.append(os.path.join(os.path.expanduser(directory),
-                                                f"{app_name}_v{app_version}_pipe_file"))
+                                                f"{app_name}_v{app_version}_{self._identifier}_pipe_file"))
```

Comment:
For extra safety we could wrap filename with `sanitize_filename()` (which replaces path separators).

Reply:
You mean this? https://pypi.org/project/sanitize-filename/

Reply:
Nope.  See Line 314 in `util/__init__.py`.

Reply:
> Nope. See Line 314 in `util/__init__.py`.

Thanks

---

So you suggest to call the sanitizer on the `os.path.join()` result, right?

Reply:
```suggestion
                                                sanitize_filename(f"{app_name}_v{app_version}_{self._identifier}_pipe_file")))
```

Reply:
I *think* he's suggesting to wrap `f"{app_name}_v{app_version}_{self._identifier}_pipe_file"`.  If you whap the whole join it would remove the path separators from the name that you're appending.

Reply:
> So you suggest to call the sanitizer on the `os.path.join()` result, right?

Nope, only on the filename, see suggestion above.